hc/api/views.py

```diff
@@ -291,6 +295,61 @@ def pings(request, code):
 
     return JsonResponse({"pings": dicts})
 
+@cors("GET")
+@csrf_exempt
+@validate_json()
+@authorize_read
+def flips(request, code):
+    check = get_object_or_404(Check, code=code)
+    if check.project_id != request.project.id:
+        return HttpResponseForbidden()
+
+    if all(x not in request.GET for x in ('start','end','seconds')):
+        flips = Flip.objects.select_related("owner").filter(
+            owner=check, new_status__in=("down","up"),
+        ).order_by("created")
+    else:
+        if any(x in request.GET for x in ('start','end')) and 'seconds' in request.GET:
+            return HttpResponseBadRequest()
+
+        history_start = None
+        history_end = datetime.now()
+
+        if 'start' in request.GET:
+            history_start = datetime.fromtimestamp(int(request.GET['start']))
+        if 'end' in request.GET:
+            history_end = datetime.fromtimestamp(int(request.GET['end']))
+
+        if 'seconds' in request.GET:
+            history_start = datetime.now()-td(seconds=int(request.GET['seconds']))
+        
+        flips = Flip.objects.select_related("owner").filter(
+            owner=check, new_status__in=("down","up"),
+            created__gt=history_start,
+            created__lt=history_end
+        ).order_by("created")
+
+    dictStatus = {"up":1,"down":0}
+
+    return JsonResponse({"flips": list(map(lambda x: {'timestamp':x.created,'up':dictStatus[x.new_status]}, flips))})
+
+    # return JsonResponse(check.to_dict(
+    #     readonly=request.readonly,
+    #     history=(
+    #         history_start,history_end
+    #         )
+    #     ))
+
+@cors("GET")
+@csrf_exempt
+@validate_json()
+@authorize_read
+def get_flips_by_unique_key(request, unique_key):
+    checks = Check.objects.filter(project=request.project.id)
```

Comment:
This will fetch a Check instance first by the unique_key, and then the second time in hc.api.views.flips by code. 

We can avoid a redundant SQL query and redundant view decorators by adding another function which both views call. Views would be responsible for locating the check (either by code or by unique_key), and the third function would be responsible for the rest: parsing timestamps, fetching flips and returning them.